msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -66,10 +107,100 @@ private void initializeSharedPreferenceFileManager(@NonNull final Context contex
+                    configuration.getAppContext(),
+                    configuration.getDefaultAuthority(),
+                    configuration) && mIsSharedDevice) {
+                getCurrentAccountFromSharedDevice(callback, configuration);
+                return;
+            }
+
+            ApiDispatcher.initializeDiagnosticContext();
+
+            com.microsoft.identity.common.internal.logging.Logger.verbose(
+                    TAG + methodName,
+                    "Getting the current account"
+            );
+
+
+            final OperationParameters params = OperationParametersAdapter.createOperationParameters(mPublicClientConfiguration);
+            final LoadAccountCommand command = new LoadAccountCommand(
+                    params,
+                    MSALControllerFactory.getAcquireTokenController(
+                            mPublicClientConfiguration.getAppContext(),
+                            params.getAuthority(),
+                            mPublicClientConfiguration
+                    ),
+                    new TaskCompletedCallbackWithError<List<ICacheRecord>, Exception>() {
+                        @Override
+                        public void onTaskCompleted(final List<ICacheRecord> result) {
+                            if (null == result || result.size() == 0) {
+                                com.microsoft.identity.common.internal.logging.Logger.verbose(
+                                        TAG + methodName,
+                                        "No account found.");
+                                checkCurrentAccountNotifyCallback(callback, null);
+                            } else {
+                                // First, transform the result into IAccount + TenantProfile form
+                                final List<IAccount>
+                                        accounts = AccountAdapter.adapt(result);
+
+                                final String trimmedIdentifier = getPersistedCurrentAccount().getHomeAccountId();
+
+                                // Evaluation precedence...
+                                //     1. home_account_id
+                                //     2. local_account_id
+                                //     3. username
+                                //     4. Give up.
+
+                                final AccountMatcher accountMatcher = new AccountMatcher(
+                                        homeAccountMatcher,
+                                        localAccountMatcher,
+                                        usernameMatcher
+                                );
+
+                                for (final IAccount account : accounts) {
+                                    if (accountMatcher.matches(trimmedIdentifier, account)) {
+                                        //callback.onTaskCompleted(account);
+                                        checkCurrentAccountNotifyCallback(callback, result);
```

Comment:
This means that as an app, I can switch back and forth between SingleAccount and MultipleAccount mode?

Reply:
That was intended.  Let's discuss why you were thinking that though.  I'm guessing I missed something.

Reply:
For some reason, I was under an impression that we were planning to 'clean the slate' every time the user switches mode. I'm also fine with this thought as it allows existing apps to switch to single account mode.